File: 프로그래머스/1/176963. 추억 점수/추억 점수.cpp



#include <bits/stdc++.h>
using namespace std;

vector<int> solution(vector<string> name, vector<int> yearning, vector<vector<string>> photo) {
    vector<int> answer;
    unordered_map<string, int> m;

    for(int i = 0; i<name.size(); i++) m[name[i]] = yearning[i];

    for(int i = 0; i<photo.size(); i++){
        int points = 0;
        for(int j = 0; j<photo[i].size(); j++){
            if(m.find(photo[i][j]) != m.end()){
                points += m[photo[i][j]];
            }
        }
        answer.push_back(points);
    }

    return answer;
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        int[] answer = new int[photo.length]; 
        Map<String, Integer> yearningMap = new HashMap<>();

        for (int i = 0; i < name.length; i++) {
            yearningMap.put(name[i], yearning[i]);
        }

        for (int i = 0; i < photo.length; i++) {
            int points = 0;
            for (String person : photo[i]) {
                points += yearningMap.getOrDefault(person, 0);
            }
            answer[i] = points;
        }

        return answer;
    }
}
